Sources/Controllers/Map/OAMapHudViewController.mm

```diff
@@ -341,31 +341,34 @@ - (BOOL) isOverlayUnderlayViewVisible
     return _overlayUnderlayView && _overlayUnderlayView.superview != nil;
 }
 
-- (void) updateOverlayUnderlayView:(BOOL)show
+- (void) updateOverlayUnderlayView
 {
-    if (!show)
-    {
-        if (_overlayUnderlayView && _overlayUnderlayView.superview)
-            [_overlayUnderlayView removeFromSuperview];
-        
-        return;
-    }
+    BOOL shouldOverlaySliderBeVisible = _app.data.overlayMapSource && [OAAppSettings sharedManager].mapSettingShowOverlayOpacitySlider;
+    BOOL shouldUnderlaySliderBeVisible = _app.data.underlayMapSource && [OAAppSettings sharedManager].mapSettingShowUnderlayOpacitySlider;
     
-    if (!_overlayUnderlayView)
+    if (shouldOverlaySliderBeVisible || shouldUnderlaySliderBeVisible)
     {
-        _overlayUnderlayView = [[OAOverlayUnderlayView alloc] init];
+        if (!_overlayUnderlayView)
+        {
+            _overlayUnderlayView = [[OAOverlayUnderlayView alloc] init];
+        }
+        
+        if (_overlayUnderlayView.viewLayout != OAViewLayoutNone && !_overlayUnderlayView.superview)
```

Comment:
Looks like new bug introduced. The _overlayUnderlayView.viewLayout changes on init / updateView methods.